### PR TITLE
additional W&B params

### PR DESCRIPTION
**Pull request by epwalsh**
Adds additional parameters to the W&B callback. See the `wandb.init` docs for reference: https://docs.wandb.ai/ref/python/init

### CODE REVIEW

**Review by dirkgr — APPROVED** [review]
Not in this PR, but I'm wondering if we should separate the experiment config from the monitoring config.

If we think of the experiment config as a record of what you ran, then configuring wandb in it is a distraction. Things that don't alter the results should not be part of the experiment config. Unless you use batch norm, how many GPUs you ran on should be irrelevant to the results, so it should not be part of the experiment config.

We might instead create a `~/.allennlp.config` file that contains this stuff.

I'd love to get some more opinions on this, @matt-gardner, @nelson-liu, @mahnerak, @wlhgtc, @pvcastro. Does that align with how you use it?

**Comment by matt-gardner** [comment]
I think you have to trade off the conceptual gain you might get from separating them with the complexity of such a configuration system.  It's another thing for a user to figure out.  I'm not sure the benefit would be worth the cost, honestly, though it's hard to say before seeing a more detailed proposal.

**Comment by pvcastro** [comment]
I do agree that unless it's something we're just experimenting with to see how it affects the model training, it would make sense to separate it from the regular training config files. From my experience, the more we can reuse in the training configs, the better. Otherwise it's just something more we have to keep in the training configs and eventually update in **all** existing configs when something in AllenNLP gets upgraded. These training callbacks are a good recent example, from my perspective.

**Comment by wlhgtc** [comment]
Maybe we could divide the whole config file into different parts: model, trainer, dataloader....
Then we just need some code to merge it in train process.

**Comment by matt-gardner** [comment]
Remember that jsonnet already gives you the ability to separate things out into another file.

**Comment by pvcastro** [comment]
Thanks @matt-gardner , I wasn't aware!
